src/arrayMethodSort.js

```diff
@@ -5,7 +5,30 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let step;
+    let callback = compareFunction;
+
+    if (!compareFunction) {
+      callback = (curr, prev) => String(curr) < String(prev) ? -1
+        : String(curr) > String(prev) ? 1 : 0;
```

Comment:
it's better not use nesting ternary. Just use `if` with `early return`

Example: 
```
if (condition) {
  return value
}

if (condition2) {
  return value2
}

return value3
```